src/useSize.tsx

```diff
@@ -16,22 +16,8 @@ const useSize = (
   element: Element,
   { width = Infinity, height = Infinity }: Partial<State> = {}
 ): [React.ReactElement<any>, State] => {
-  if (!isBrowser) {
-    return [
-      typeof element === 'function' ? element({ width, height }) : element,
-      { width, height },
-    ];
-  }
```

Comment:
IMO, the browser check should stay at the top. Event better would be if we didn't have the browser check in this hook, but instead had two separate hooks

```ts
const useSize = isBrowser ? useSizeBrowser : useSizeServer;
export default useSize;
```

Reply:
Good idea. I splitted useSize to useSizeBrowser and useSizeServer